src/test/java/ru/hogwarts/school/controller/StudentControllerMockMvcTest.java

```diff
@@ -0,0 +1,139 @@
+
+    @SpyBean
+    private StudentServiceImpl studentService;
+
+    @MockBean
+    private StudentRepository studentRepository;
+
+    @InjectMocks
+    private StudentController studentController;
+
+    @Test
+    public void testFindStudentById() throws Exception {
+        Student student = new Student(1L, "name", 34);
+
+        when(studentService.getStudentById(1L)).thenReturn(student);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/student/1")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+
+    @Test
+    public void testCreateStudent() throws Exception {
+        JSONObject studentObject = new JSONObject();
+        studentObject.put("id", 1L);
+        studentObject.put("name", "name");
+        studentObject.put("age", 34);
+
+        Student student = new Student();
+        student.setName("name");
+        student.setAge(34);
+
+        when(studentRepository.save(any(Student.class))).thenReturn(student);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .post("/student")
+                        .content(studentObject.toString())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name").value("name"))
+                .andExpect(jsonPath("$.age").value(34));
+    }
+    @Test
+    public void testFindStudentsByAgeBetween() throws Exception {
+        JSONObject studentObject = new JSONObject();
+        studentObject.put("id", 1L);
+        studentObject.put("name", "name");
+        studentObject.put("age", 34);
+
+        Student student = new Student();
+        student.setName("name");
+        student.setAge(34);
+
+        JSONObject studentObject1 = new JSONObject();
+        studentObject.put("id", 2L);
+        studentObject.put("name", "bob");
+        studentObject.put("age", 24);
+
+        Student student1 = new Student();
+        student.setName("bob");
+        student.setAge(24);
+
+
+        when(studentRepository.save(any(Student.class))).thenReturn(student, student1);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .post("/student")
+                        .content(studentObject.toString())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/student/byAgeBetween?max=35&min=23")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+
+    }
+
+    @Test
+    public void testGetFacultyByStudent() throws Exception {
+
+        Student student = new Student(1L, "name", 34);
+
+        when(studentService.getStudentById(1L)).thenReturn(student);
```

Comment:
```suggestion
        when(studentRepository.findById(1L)).thenReturn(Optional.of(student));
```
Мокать нужно репозиторий